docs/interfaces.md

```diff
@@ -0,0 +1,568 @@
+        bytes chaincode_EK   = 2;       // public chaincode encryption key
+        bytes cckeys_enc     = 3;       // chaincode keys encrypted for the receiver
+        bytes enclave2_vk    = 4;       // receiver of this message
+        bytes enclave_vk     = 5;       // creator of this message
+}
+
+message SignedExportMessage {
+        bytes export_msg_bytes = 1;
+        bytes signature        = 2;     // signature of the message creator
+}
+
+message AttestationEvidence {
+        // this message is defined with more detail in #412
+        // https://github.com/hyperledger-labs/fabric-private-chaincode/issues/412
+        enum Type {
+                SGX_EPID = 0;
+                SGX_DCAP = 1;
+                WHATEVER = 2;
+        }
+        Type type     = 1;
+        bytes payload = 2;
+}
+
+```
+
+ERCC keeps in instance of an attestation.Verifier to check an attestation evidence message. ERCC just passes the serialzed attestation evidence message to the verifier. 
+Depending on the attestation protocol (e.g., EPID- or DCAP-based attestation), the verifier implementes the corresponding logic. Details of the evidence verification are defined in [#412](https://github.com/hyperledger-labs/fabric-private-chaincode/issues/412).
+
+```go
+type EnclaveRegistryCC struct {
+	ra  attestation.Verifier
+}
+
+```
+
+
+# TLCC
+
+Some notes on TLCC. The TLCC instance is currently implemented as a systemchaincode, that is, there exists only a single instance. As a peer can participate in many channels, there is a seperate TLCC_enclave per each channel. The TLCC chaincode is responsible to multiplex requests for a given channel to the corresponding TLCC_enclave. Note for MVP, we only support a single channel but the interface specified here should already take multi-channel support into account.
+
+## Interface:
+The TLCC interface is implemented through a "normal" chaincode interface.
+
+```go
+// chaincode interface
+func Invoke(stub shim.ChaincodeStubInterface) pb.Response {}
+
+// functions below are dispatched by Invoke implementation
+
+// used by the admin to join the channel
+// Note: this methods creates a new tlcc_enclave and starts a block listener
+// for the given channel_id that passes the genesis block and all
+// subsequent blocks to the tlcc_enclave.
+func joinChannel(channel_id string) error {}
+```
+
+```c++
+// Provided/implemented by a common logging module
+// interface exposed to TLCC_enclave via cgo
+void ocall_print_string([in, string] const char *str);
+```
+
+```go
+// interface to establish/close a secure channel between an
+// ECC_Enclave and a TLCC_enclave.
+func session_setup(setup SessionSetup) (SessionSetupResponse, error) {}
+func session_close(close SessionClose) (SessionCloseReponse, error) {}
+
+// interface to invoke a TLCC request, such as, getMetadata and validateMSPId
+func session_request(request SessionRequest) (SessionRequestResponse, error) {}
+```
+
+## State:
```

Comment:
Below is not really "State"?   The state tlcc would have is essentially just a lookup table of channel-id->tlcc enclave?

Reply:
good point!

Reply:
added

Reply:
you addressed my second comment, but not the first: the "State" heading should really be moved down just in front of the newly added map; the protobuf are really part of the interface section?

Reply:
aha, heading moved down.